internal/controller/buildpipeline/buildpipeline_adapter.go

```diff
@@ -454,3 +523,125 @@ func (a *Adapter) addPRGroupToBuildPLRMetadata(pipelineRun *tektonv1.PipelineRun
 	a.logger.Info("can't find source branch info in build PLR, not need to update build pipelineRun metadata")
 	return nil
 }
+
+// prepareFakeSnapshot will create a snapshot object to copy the labels/annotations from build pipelinerun
+// and be used to communicating with git provider
+func (a *Adapter) prepareFakeSnapshot(pipelineRun *tektonv1.PipelineRun) *applicationapiv1alpha1.Snapshot {
```

Comment:
Wdyt about naming this something like `prepareEmptySnapshot` or `prepareTemporarySnapshot`?

Reply:
Updated to use `prepareTemporarySnapshot`.  Thanks.